add cidr to glossary (#17550)
kubernetes/website: content/en/docs/reference/glossary/cidr.md

---
title: CIDR
id: cidr
date: 2019-11-12
full_link: 
short_description: >

>   CIDR is a notation for describing blocks of IP addresses and is used heavily in various networking configurations.

Reviewer: ```suggestion
  A notation for describing blocks of IP addresses
```

Reviewer: This could be a glossary tooltip. I think these work well when succinct; someone who wants to learn more can click.

aka:
tags:
- networking
---

> CIDR ( Classless Inter-Domain Routing) is a notation for describing blocks of IP addresses and is used heavily in various networking configurations.

Reviewer: ```suggestion
 CIDR (Classless Inter-Domain Routing) is a notation for describing blocks of IP addresses and is used heavily in various networking configurations.
```

Reviewer: but see my other comment

<!--more-->

> In the Kubernetes concept, each node is assigned a range of IP addresses through the start address and a netmask. This allows nodes to assign each <Pod> a unique IP address.

Reviewer 2: ```suggestion
In the context of Kubernetes, each {{< glossary_tooltip text="Node" term_id="node" >}} is assigned a range of IP addresses through the start address and a subnet mask using CIDR. This allows Nodes to assign each {{< glossary_tooltip text="Pod" term_id="pod" >}} a unique IP address. Although originally a concept for IPv4, CIDR has also been expanded to include IPv6. 
```

> It is an IP addressing scheme that replaces the older system based on classes A, B, and C. With CIDR, a single IP address can be used to designate many unique IP addresses. A CIDR IP address looks like a normal IP address except that it ends with a slash followed by a number, called the IP network prefix.

Reviewer: That's the correct meaning of [CIDR](https://tools.ietf.org/rfc/rfc1817.txt) generally. However, Kubernetes uses _CIDR_ to mean “a block of IP addresses that you can describe with an start address and a netmask”. For example, various IPv6 settings use the term CIDR even though CIDR itself is purely an IPv4 concept.

Reviewer 2: To that end, maybe something add something specific to how it is used in Kubernetes, like `In Kubernetes, each node is assigned a range of IP addresses through the start address and a netmask. This allows nodes to assign each <Pod> a unique IP address.`

> In the context of Kubernetes, each {{< glossary_tooltip text="Node" term_id="node" >}} is assigned a range of IP addresses through the start address and a subnet mask using CIDR. This allows Nodes to assign each {{< glossary_tooltip text="Pod" term_id="pod" >}} a unique IP address. Although originally a concept for IPv4, CIDR has also been expanded to include IPv6. 

Reviewer: Kubernetes uses CIDR blocks for various things including this example.
`ipBlock` from NetworkPolicy also refers to blocks of IP addresses.

Is it helpful to show an actual example, eg mention 203.0.113.128/25 and show the range that it covers?

Reviewer 2: Ok, maybe something then to the effect of: "CIDR is used in various networking aspects in Kubernetes. For example..."

I'm not sure providing an example would be useful, in my experience CIDR isn't super intuitive to people who have never seen it before so the example would probably be lost on them. I think there would be two main audiences for this glossary: people who don't know what CIDR is or people who know what CIDR is, but want to know how it applies to Kubernetes. Right now I think the definition helps more of the second audience and the discussion is how much should we cater towards the first audience (complete beginners to CIDR)

Reviewer: How does SIG Docs feel about hyperlinking to, eg, Wikipedia's page about this?

Reviewer 3: My feeling is that we don't need to dive into the details of CIDR which is pretty common an acronym in the community.
